modules/gapi/include/opencv2/gapi/cpu/gcpukernel.hpp

```diff
@@ -127,16 +128,28 @@ class GAPI_EXPORTS GCPUContext
 class GAPI_EXPORTS GCPUKernel
 {
 public:
-    // This function is kernel's execution entry point (does the processing work)
-    using F = std::function<void(GCPUContext &)>;
+    // This function is a kernel's execution entry point (does the processing work)
+    using RunF = std::function<void(GCPUContext &)>;
+    // FIXME: Workaround to hide dangerous bug:
+    //        If we add callbacks here as std::function-s then we will get
+    //        into segfault with "stack smashing detected" error for GPU tests.
+    //        Segfault will be happened on GOCLBackendImpl::unpackKernel exit.
+    //        But, if we comment last statement in GOCLBackendImpl::unpackKernel
+    //        so this issue will not be reproducible.
+    // This function is a stateful kernel's setup routine (configures state)
+    using SetupF = void(*)(GMetaArgs, GArgs, GArg &);
+    // This function is a stateful kernel's execution entry point (does the processing work)
+    using StRunF = void(*)(GCPUContext &, GArg &);
 
     GCPUKernel();
-    explicit GCPUKernel(const F& f);
+    explicit GCPUKernel(const RunF& runF);
+    explicit GCPUKernel(const SetupF& setupF, const StRunF& stRunF);
```

Comment:
There's no point in `explicit` for this constructor.

Reply:
Made it is implicit